### DISCUSSION TITLE
Budget period oddity

### DISCUSSION THREAD

**Discussion by BackSeat** [firefly-iii/firefly-iii#8145] [Firefly III]
I'm seeing some odd dates in my budgets.

I'd like to budget annually between 1 November and 31 October. I've set the Options/Preferences/Fiscal year start date to 1/11/2023. I've set up a number of budgets, all of them annual.

When I look at the Budgets page, there three things that appear inconsistent:

- Almost all of the budget periods are shown as "January 1st, 2023 to December 31st, 2023" rather than the November - October dates I want.
- ...except for one, which is shown as "November 1st, 2023 to November 30th, 2023". I have checked, and it is definitely a yearly budget:

![image](https://github.com/firefly-iii/firefly-iii/assets/827233/d5d36ed2-8f85-4c50-9b0c-ca9be854fb8f)

- At the foot of the budget page where expenses without a budget are listed, it says, "Expenses without budget between November 1st, 2023 and October 31st, 2024", which _is_ the dates I want.
- When I look at the budget report, fiscal year, it shows me the dates I expect.

So there seems to be some inconsistency in the way budgets are displayed. 

What more information can I give to help resolve this?

**Comment by JC5** [comment]
Yearly budgets are always year-to-year, the fiscal year doesn't change that. I have it on the list somewhere, but it's not really high up.

**Reply by BackSeat** [reply]
Thanks. I'm happy to look at what's needed to fix that - would you accept a PR? Expectation setting: it won't be done that soon.

Any comment on the oddity of the Household budget period?

**Reply by JC5** [reply]
Adding custom budget periods is a nightmare to maintain, so I'm more concerned about that than anything else. So no, no PR's please. I hope to fix this together with the new layout.

The budget overview page follows the date period at the top of the period, that's the reason it's like that. I have no idea why the Household budget period is like that, I can't reason backwards about that.

A trick to set the budget periods as you want them is by setting the period selector on the top of the page to your custom yearly period, and then enter the amounts. 

Sorry, this isn't yet the way I'd like it to be.
